pkg/controller/migmigration/annotation.go

```diff
@@ -170,22 +192,30 @@ func (t *Task) labelNamespaces(client k8sclient.Client) error {
 			},
 			&namespace)
 		if err != nil {
-			return liberr.Wrap(err)
+			return itemsUpdated, liberr.Wrap(err)
 		}
 		if namespace.Labels == nil {
 			namespace.Labels = make(map[string]string)
 		}
+		if _, exist := namespace.GetLabels()[IncludedInStageBackupLabel]; exist {
+			continue
+		}
 		namespace.Labels[IncludedInStageBackupLabel] = t.UID()
 		err = client.Update(context.TODO(), &namespace)
 		if err != nil {
-			return liberr.Wrap(err)
+			return itemsUpdated, liberr.Wrap(err)
 		}
 		log.Info(
 			"NS annotations/labels added.",
 			"name",
 			namespace.Name)
+		itemsUpdated++
+		if itemsUpdated > 50 {
+			t.setProgress([]string{fmt.Sprintf("%v/%v Namespaces labeled", i, total)})
```

Comment:
@pranavgaikwad can you confirm this use of `t.setProgress` is correct? It looks like this is wiping out any existing progress and creating a new progress string array. 

In the next function within this same phase the same thing will happen with a different progress message. 
```
t.setProgress([]string{fmt.Sprintf("%v/%v Pod annotations/labels added.", i, total)})
```

Reply:
@djwhatle if `t.setProgress` is used once in a single reconcile it should be fine. And that is what will happen here always since we are requeuing right after we set to progress. thoughts? @pranavgaikwad . We can also pass `progress` array between different functions just like `lastUpdate`, but I am not sure if that is useful. any thoughts?

Reply:
I think this is OK.